test/viewmodels/challenge_view_model_integration_test.dart

```diff
@@ -193,6 +183,47 @@ void main() {
       final points = updatedUser.data.centauriPoints;
       expect(points, ChallengeRepositoryService.soloChallengeReward);
     });
+
+    test("Challenges position is updated on user position change and refresh",
+        () async {
+      final posts = postGenerator.generatePostsAt(userPosition1, 3);
+      setPostsFirestore(posts, fakeFireStore);
+
+      final activeChallenges =
+          challengeGenerator.generateChallenges(1, false, extraTime);
+
+      await setChallenges(
+        fakeFireStore,
+        activeChallenges,
+        testingUserFirestoreId,
+      );
+
+      var challenges = await container.read(challengeViewModelProvider.future);
+
+      // Check initial distances are 0 since user is at userPosition1
+      for (var challenge in challenges) {
```

Comment:
I think that we can have a `final` instead of a `var` here :+1: 
```suggestion
      for (final challenge in challenges) {
```

(The same applies for the other `for` loop :) )

Reply:
Fixed in 66ff5e1c962070c580d9b3943e0a9e6d5a61cb11